src/v/datalake/record_translator.cc

```diff
@@ -188,6 +189,11 @@ structured_data_translator::build_type(std::optional<resolved_type> val_type) {
     if (val_type.has_value()) {
         val_id = std::move(val_type->id);
         auto& struct_type = std::get<iceberg::struct_type>(val_type->type);
+        // TODO: could make this configurable at some point
+        std::ignore = iceberg::for_each_field(
+          struct_type, [](iceberg::nested_field* f) {
+              f->required = iceberg::field_required::no;
+          });
```

Comment:
> nested fields

This is essentially all fields right? Because the type root is a struct type correct?

Reply:
yup